### ISSUE TITLE
Files open in new window instead of new tab

### ISSUE THREAD

**Issue by nadrolinux** [flathub/com.visualstudio.code#75]
Hello,
Current version doesn't work properly in Ubuntu. When I try to open file by a right click action in Nautilus, file is opened in new window (new instance) instead of a new tab (share one VC instance).
Cheers,

**Comment by thiblahute** [comment]
This one is quite annoying indeed.

**Comment by subins2000** [comment]
Yeah, this is annoying. Please fix

**Comment by eszlari** [comment]
Maybe a workaround from the discord flatpak could be used:
https://github.com/flathub/com.discordapp.Discord/wiki/Rich-Precense-(discord-rpc)
https://github.com/flathub/com.discordapp.Discord/blob/master/discord.sh

**Comment by gasinvein** [comment]
Nah, Discord has persistent socket name, while VS Code generates it from filesystem path hash and changes with each version, we can't pre-determine it reliably.

In best case scenario, VS Code should be patched somehow [like this](https://github.com/gasinvein/flatpak-apps-staging/blob/b77cb67ce25204aa3f2e963440190ec763b6ec57/patches/ipc-sockets-paths-flatpak.patch),  but it's obviously impossible to apply patches downstream on proprietary version.

**Comment by eszlari** [comment]
>  we can't pre-determine it reliably.

What about creating the link after the app has started?

**Comment by gasinvein** [comment]
> What about creating the link after the app has started?

I'm afraid it'll be even more fragile than with Discord. We need to wait for the app to exit and terminate `socat` after it, but VS Code generally forks to background on launch. I don't know any easy way to watch forked child processes (except for creating cgroups which is unavailable in flatpak sandbox).

**Comment by gasinvein** [comment]
FYI this could be fixed by flatpak/flatpak#4126

**Comment by WiliTest** [comment]
Same problem on windows (since the begining): when I click on a file (eg. when I click on a .txt vscode opens it on a new session instead of opening in a new tab)
